ls1intum/Artemis
# Add support for importing students to an exam via login

### Checklist

### Motivation and Context

Fixes #2517.

### Description

A new row `LOGIN` was added to the possible CSV options. Now either that or `REGISTRATION_NUMBER` must be present to register a student for the exam.

### Steps for Testing

1. Go to an existing exam or create a new one
2. Click on "Students" and then "Import students"
3. Upload a CSV file and import it

Example CSV file layouts:
```
REGISTRATION_NUMBER
<number>
```
Where `<number>` is to be replaced by a registration number (matr. nr.). If the number was a valid one, the row should show as green after import. If the number is invalid the row should be red.

```
LOGIN
<login>
```
Where `<login>` is to be replaced by a user login. If the login was a valid one with a registration number (note: accounts such as `artemis_admin` might not have a matr. number), the row should show as green after import. If it was invalid the row should be red.

A mixed case which combines both examples above into one with two students to import:
```
REGISTRATION_NUMBER,LOGIN
<number>,
,<login>
```

The following column names are interchangeable for registration numbers: `REGISTRATION_NUMBER`, `MATRIKEL_NUMMER`, and `MATRICULATION_NUMBER`
For logins: `LOGIN`, `USERNAME`, `USER`, `BENUTZER`, and `BENUTZERNAME`
Further possible columns (which are not required and don't work on their own though) are `FIRST_NAME_OF_STUDENT`, `FAMILY_NAME_OF_STUDENT`, `FIRST_NAME`, `LAST_NAME`, and `FAMILY_NAME`.

All names can include underscores and whitespaces. They can be lower case or mixed case too. Be sure to test a combination of names with different case, e.g. `Matrikel Nummer`:
```
Matrikel Nummer,Benutzer
<number>,
,<login>
```

### Screenshots
When selecting a file:
![grafik](https://user-images.githubusercontent.com/72132281/104573812-a37cb500-5655-11eb-8a3f-4404eee0b622.png)

New text and tooltips:
![grafik](https://user-images.githubusercontent.com/72132281/104594356-53135080-5671-11eb-9186-b4e22b8a05d7.png)
![grafik](https://user-images.githubusercontent.com/72132281/104595339-c10c4780-5672-11eb-915f-7f1515aa6b4e.png)
![grafik](https://user-images.githubusercontent.com/72132281/104595374-cf5a6380-5672-11eb-87db-ca704dd4c3b9.png)
![grafik](https://user-images.githubusercontent.com/72132281/104595393-d84b3500-5672-11eb-95c3-6b0ef9f46f17.png)
